--- papers-48.5/shell/src/meson.build.orig	2025-07-20 09:36:03.000000000 +0200
+++ papers-48.5/shell/src/meson.build	2025-08-14 19:10:41.813471440 +0200
@@ -10,6 +10,7 @@ source_root = meson.current_source_dir()
 cargo = find_program('cargo', required: true)
 
 cargo_options = [ '--manifest-path', shell_rs_root / 'Cargo.toml' ]
+cargo_options += [ '--target', 'x86_64-unknown-linux-gnux32' ]
 cargo_options += [ '--target-dir', meson.current_build_dir() ]
 
 if get_option('optimization') == '0'
@@ -75,7 +76,7 @@ custom_target(
   output: meson.project_name(),
   command: [
     'cp',
-    meson.current_build_dir() / rust_target / meson.project_name(),
+    meson.current_build_dir() / 'x86_64-unknown-linux-gnux32' / rust_target / meson.project_name(),
     '@OUTPUT@',
   ],
 )
